### PR TITLE
Write the name of the policy into iptables as a comment.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Makes it easier to debug policy.
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
When generating iptables chains for network policies, Felix now adds a comment to the first rule in the chain with the name of the policy.
```

### CODE REVIEW

File: rules/policy.go
Comment (neiljerram) [inline rules/policy.go:55]: Why allowing for multiple comments, since it looks like callers only provide one.

File: rules/policy.go
Comment (neiljerram) [inline rules/policy.go:61]: Does this case arise in practice?  I'd want to double check that adding the following comment-only rule doesn't change any behaviour.

**Review by neiljerram — COMMENTED** [review]
Nice!  Couple of queries.

File: rules/policy.go
Comment (fasaxc) [reply rules/policy.go:55]: It was mainly to make it an optional argument so that the existing tests could stay as-is and avoid conflicts when merging over to private.

File: rules/policy.go
Comment (fasaxc) [reply rules/policy.go:61]: Yes, it does arise. A rule with just a comment is a no-op, packets just fall through to whatever would happen next.

Having said that, my change causes a panic due to having a nil action on the rule.  I'll fix that up.

**Review by neiljerram — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by fasaxc** [comment]
Haha, guess that FV isn't actually BPF safe :grin:

**Comment by alban** [comment]
Thanks! Does it also add a comment for GlobalNetworkPolicies?

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by fasaxc** [comment]
> Thanks! Does it also add a comment for GlobalNetworkPolicies?

Yes, there's only one type of policy by this point in the processing.

**Pull request merged**
